File: src/main/java/de/cplaiz/activecraftcore/messages/MessageSupplier.java
package de.cplaiz.activecraftcore.messages;

import org.bukkit.ChatColor;

public record MessageSupplier(ActiveCraftMessage activeCraftMessage, Language language) {

    public String getMessage(String input) {
        return activeCraftMessage.getMessage(input);
    }

    public String getMessage(String input, ChatColor color) {
        return activeCraftMessage.getMessage(input, color, language);
    }

    // raw msg
    public String getRawMessage(String input) {
        return activeCraftMessage.getRawMessage(input);
    }

    // format
    public String getFormatted(String key, MessageFormatter formatter) {
        return activeCraftMessage.getFormatted(key, formatter);
    }


    public String getFormatted(String key, MessageFormatter formatter, ChatColor color) {
        return activeCraftMessage.getFormatted(key, formatter, color);
    }

}

File: src/main/java/org/activecraft/activecraftcore/messages/MessageSupplier.kt
package org.activecraft.activecraftcore.messages

import org.activecraft.activecraftcore.messages.messagecollections.Durations
import org.activecraft.activecraftcore.messages.messagecollections.Errors
import org.activecraft.activecraftcore.messages.messagecollections.Reasons
import org.activecraft.activecraftcore.playermanagement.Profile
import org.bukkit.ChatColor
import org.bukkit.command.CommandSender

fun CommandSender.getMessageSupplier(acm: ActiveCraftMessage): MessageSupplier {
    val profile = Profile.of(this) ?: return acm.getDefaultMessageSupplier()!!
    return profile.getMessageSupplier(acm)
}

class MessageSupplier(val activeCraftMessage: ActiveCraftMessage, val language: Language) {

    val colorScheme: ColorScheme
        get() = activeCraftMessage.colorScheme
    val errors = Errors(this)
    val durations = Durations(this)
    val reasons = Reasons(this)

    fun getMessage(key: String, color: ChatColor? = colorScheme.primary) =
        activeCraftMessage.getMessage(key = key, color = color, language = language)

    fun getRawMessage(key: String) =
        activeCraftMessage.getRawMessage(key = key, language = language)

    fun getFormatted(key: String, formatter: MessageFormatter, color: ChatColor? = colorScheme.primary) =
        activeCraftMessage.getFormatted(key = key, color = color, formatter = formatter, language = language)
}